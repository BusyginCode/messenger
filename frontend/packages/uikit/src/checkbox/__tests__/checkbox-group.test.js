import React from 'react'
import { shallow } from 'enzyme'
import Checkbox from '../checkbox'
import CheckboxGroup from '../checkbox-group'

describe('uikit/checkbox-group', () => {
  it('should renders correctly', () => {
    const onChange = jest.fn()
    const onBlur = jest.fn()
    const onFocus = jest.fn()
    const options = [
      {
        value: 'val1',
        label: 'Value1',
        iconLeft: 'left',
        iconTop: 'top',
        imageLabel: 'label',
        iconRight: 'right'
      },
      { value: 'val2', label: 'Value2', disabled: true }
    ]
    const wrapper = shallow(
      <CheckboxGroup
        max={2}
        options={options}
        value={['val1', 'val2']}
        onChange={onChange}
        onBlur={onBlur}
        onFocus={onFocus}
      />
    )

    expect(wrapper).toMatchSnapshot()

    wrapper
      .find(Checkbox)
      .at(0)
      .prop('onBlur')({ target: { value: 'val1' } })

    wrapper
      .find(Checkbox)
      .at(0)
      .prop('onFocus')({ target: { value: 'val1' } })

    wrapper
      .find(Checkbox)
      .at(0)
      .prop('onChange')({ target: { value: 'val1' } })

    wrapper.setProps({ value: [] })

    wrapper
      .find(Checkbox)
      .at(0)
      .prop('onChange')({ target: { value: 'val1' } })

    wrapper.setProps({ radio: true, value: null })

    wrapper
      .find(Checkbox)
      .at(0)
      .prop('onChange')({ target: { value: 'val1' } })

    wrapper.setProps({ checkboxRadio: true })

    wrapper
      .find(Checkbox)
      .at(0)
      .prop('onChange')({ target: { value: 'val1' } })

    wrapper.setProps({ value: ['val1'] })

    wrapper
      .find(Checkbox)
      .at(0)
      .prop('onChange')({ target: { value: 'val1' } })

    expect(onChange).toHaveBeenCalledTimes(5)
  })

  it('should renders correctly disabled=true', () => {
    const onChange = jest.fn()
    const options = [{ value: 'val1', label: 'Value1' }, { value: 'val2', label: 'Value2' }]
    const wrapper = shallow(
      <CheckboxGroup
        disabled
        options={options}
        onChange={onChange}
        iconLeft="left"
        iconTop="top"
        imageLabel="label"
        iconRight="right"
      />
    )

    expect(wrapper).toMatchSnapshot()
  })
})
