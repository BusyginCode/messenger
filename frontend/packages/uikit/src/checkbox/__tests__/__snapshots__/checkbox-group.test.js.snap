// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`uikit/checkbox-group should renders correctly 1`] = `
<div
  className="checkboxGroup inline"
>
  <label
    className="checkboxWrapper checkboxWrapperColumn active"
    data-role="checkbox_group_val_1"
    key="val1"
  >
    <div
      className="icon"
      data-role="checkbox_group_icon_left_val_1"
    >
      left
    </div>
    <div
      className="icon"
      data-role="checkbox_group_icon_top_val_1"
    >
      top
    </div>
    <Checkbox
      disabled={false}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      value={true}
    />
    <div
      className="icon"
      data-role="checkbox_group_image_label_val_1"
    >
      label
    </div>
    <div
      className="icon"
      data-role="checkbox_group_icon_right_val_1"
    >
      right
    </div>
    <AText
      breed="body"
      className="label"
      data-role="checkbox_group_label_value_1"
      semibold={true}
      size="l"
    >
      Value1
    </AText>
  </label>
  <label
    className="checkboxWrapper active disabled"
    data-role="checkbox_group_val_2"
    key="val2"
  >
    <Checkbox
      disabled={true}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      value={true}
    />
    <AText
      breed="body"
      className="label"
      data-role="checkbox_group_label_value_2"
      semibold={true}
      size="l"
    >
      Value2
    </AText>
  </label>
</div>
`;

exports[`uikit/checkbox-group should renders correctly disabled=true 1`] = `
<div
  className="checkboxGroup inline disabled"
>
  <label
    className="checkboxWrapper disabled"
    data-role="checkbox_group_val_1"
    key="val1"
  >
    <Checkbox
      disabled={true}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      value={false}
    />
    <AText
      breed="body"
      className="label"
      data-role="checkbox_group_label_value_1"
      semibold={false}
      size="l"
    >
      Value1
    </AText>
  </label>
  <label
    className="checkboxWrapper disabled"
    data-role="checkbox_group_val_2"
    key="val2"
  >
    <Checkbox
      disabled={true}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      value={false}
    />
    <AText
      breed="body"
      className="label"
      data-role="checkbox_group_label_value_2"
      semibold={false}
      size="l"
    >
      Value2
    </AText>
  </label>
</div>
`;
